tensorflow/stream_executor/dnn.h

```diff
@@ -667,8 +667,31 @@ class PoolingDescriptor {
   std::vector<int64> strides_;
 };
 
-typedef int64 AlgorithmType;
-constexpr AlgorithmType kDefaultAlgorithm = -1;
+class AlgorithmType {
```

Comment:
Please add a comment explaining what this is.

Reply:
In particular, it seems that "type" here does not refer to "C++ type", but instead "type of algorithm".  That was confusing to me; probably worth calling that out explicitly in a comment.

Reply:
Renamed dnn::AlgorithmType to dnn::AlgorithmDesc and added explanatory comment.